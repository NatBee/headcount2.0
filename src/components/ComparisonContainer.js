import React, { Component } from 'react';
import Card from './Card';
import '../styles/ComparisonContainer.css';

const ComparisonContainer = (props) => {

  const dataArray = Object.keys(props.cardCompData)
  const newCard = dataArray.map((location, index) => <Card data={ props.cardCompData[location]} key={index}/>)

  return (
    <section className='comparison-cards'>
      <h1>Comparison Section</h1>
        <article className='comp-card-display'>
          <h2 className='comp-card'>{newCard[0]}</h2>
          <h2>Comparison</h2>
          <h2 className='comp-card'>{newCard[1]}</h2>
        </article>
    </section>
  )
}

export default ComparisonContainer