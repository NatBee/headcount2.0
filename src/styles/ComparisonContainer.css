

.comparison-cards {
  text-align: center;
  width:100vw;
  border-bottom: 1px solid grey;
  position: fixed;
  opacity: 1;
  transition: all .3s;
}

.comparison-cards.hide {
  transform: translateY(-100vw);
  opacity: 0;
}

.comparison-cards.show {
  transform: translateY(0);
  display: block;
  background-color: rgba(0, 0, 0, 0.85);
}
Â 

.comparison-cards h1 {
  color: #7087A8;
  font-size: 2em;
  margin: 0;
  padding: 25px 0px 0px;
  font-family: 'Playfair Display', serif;

}

.comp-card {
  font-size: 1em;
  font-weight: normal;
}

.comp-card-display {
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}

.close-compare{
  position: absolute;
  left: 0;
}

.drag-prompt{
  position: absolute;
}

.comparison {
  color: #ffff;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

