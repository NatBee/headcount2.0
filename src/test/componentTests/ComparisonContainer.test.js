import React from 'react';
import ReactDOM from 'react-dom';
import ComparisonContainer from '../../components/ComparisonContainer';
import { shallow } from 'enzyme';

it('should match the snapshot', () => {
  const renderedComponent = shallow(<ComparisonContainer/>)
  expect(renderedComponent).toMatchSnapshot()
})