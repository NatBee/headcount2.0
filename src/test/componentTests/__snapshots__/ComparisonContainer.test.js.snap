// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Comparison area
</h2>,
      "className": "comparison-wrap",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Comparison area",
      },
      "ref": null,
      "rendered": "Comparison area",
      "type": "h2",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Comparison area
</h2>,
        "className": "comparison-wrap",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Comparison area",
        },
        "ref": null,
        "rendered": "Comparison area",
        "type": "h2",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
